### DISCUSSION TITLE
Export API rules/filters for consistency between dev and prod environments

### DISCUSSION THREAD

**Discussion by MeharGaur** [pocketbase/pocketbase#232] [Q&A] [answered]
Hi, 

I have a development pocketbase instance running on localhost and I have a production instance running on a VM. So far my setup works fine, I just make a build and then use the `scp` command to copy the binary to the prod server and then restart the service and NGINX. However two things need to be considered- database schema and configuration (rules, settings, etc).

For copying the database schema, pocketbase has migrations and there's also the sqlite3 .schema command. 

However there's no clear way to copy over API rules/filters, settings, etc, without either
1) copying over the entire `pb_data` directory (which is not ideal because it holds dev values for things like S3 and SMTP).
2) doing each change one by one manually in the admin dashboard for both dev and prod.

I am thinking one way would be to write a node script that uses the Settings Web API and copies it like that, but there's no Web API for rules. Is there a better way to go about making dev and prod pocketbase instances consistent?

If we could add a section in the docs for a clear workflow for deploying from dev to prod that would help a lot with adoption 👍

**Comment by ganigeorgiev** [comment, marked as answer]
> but there's no Web API for rules

Collection API rules are part of the collections api (https://pocketbase.io/docs/api-collections/) and if you want, you can go with your approach for using the API to sync the dev and prod environments.

The issue has some nuances depending on how you use PocketBase:
- using a prebuilt PocketBase executable
- using PocketBase as framework

#### Using a prebuilt PocketBase executable
We are limited here more-or-less in terms of automation.
The only easy way (at least that I can think of) to migrate your collection schema and API rules would be to have "Import/Export" functionality from the Admin UI. This is already in the [roadmap](https://github.com/orgs/pocketbase/projects/2) and will be implemented as part of https://github.com/pocketbase/pocketbase/issues/48.

#### Using PocketBase as framework
When you are using PocketBase as framework the proper way to approach this problem is to use DB migrations.
This is already discussed in https://github.com/pocketbase/pocketbase/discussions/175 and in order to improve the dev experience we will have to:
- add a console command that will generate a migration file(s) with your current collections state automatically
- add more advanced documentation, guides and examples how to use PocketBase as framework

All of this is in the [roadmap](https://github.com/orgs/pocketbase/projects/2) and I'll start working on it in the near future (next week or the week following it).

Additional suggestions how to improve the development and deployment workflow are more than welcome.

---

_I'll move the issue to discussion since the format is more suitable._
